### PR TITLE
feat(Geotastic): add presence

### PR DESCRIPTION
**Pull request by AzNate**
## Description 
A new presence for the geography location guessing game, Geotastic. Detects all pages and displays info for all game modes.

## Acknowledgements
✓ I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
✓ I linted the code by running `yarn format`

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image
</details>

### CODE REVIEW

File: websites/G/Geotastic/presence.ts [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+										break;
+									}
+									case "/account/quota-consumption": {
+										presenceData.details = "Viewing their quota consumption";
+										break;
+									}
+									case "/account/account-settings":
+										{
+											presenceData.details = "Viewing their account settings";
+											// No default
```
Comment (EncryptedDev) [inline websites/G/Geotastic/presence.ts:138]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Comment by EncryptedDev** [comment]
Awaiting @Kozou4ever approval

File: websites/G/Geotastic/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "Nate",
+		"id": "196817730257551360"
+	},
+	"service": "Geotastic",
+	"description": {
+		"en": "Geotastic is a geographical location guessing game created by Eduard But from Edutastic Games.",
+		"fr": "Geotastic est un jeu de devinette de emplacement géographique créé par Eduard But de Edutastic Games."
```
Comment (Kozou4ever) [inline websites/G/Geotastic/dist/metadata.json:10]: This translation would be the most appropriate in my opinion 
```suggestion
		"fr": "Geotastic est un jeu de devinette de localisation géographique créé par Eduard But de Edutastic Games."
```

**Review by itsmelouis — CHANGES_REQUESTED** [review]

File: websites/G/Geotastic/presence.ts [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+				document.querySelector(".own-score").textContent
+			}`;
+		} else if (document.querySelector(".active-component-finished")) {
+			presenceData.details = "Round finished";
+			presenceData.state = `Final score ${
+				document.querySelector(".score").textContent
+			}`;
+		} else presenceData.details = "Playing a game";
+	} else {
+		switch (document.location.pathname) {
```
Comment (ImRodry) [inline websites/G/Geotastic/presence.ts:51]: why can't this start at the top?

File: websites/G/Geotastic/presence.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+				presenceData.details = "Creating a challenge";
+				break;
+			}
+			default:
+				if (document.location.pathname.includes("/challenge-details/"))
+					presenceData.details = "Viewing a challenge";
+				else if (document.location.pathname.includes("/online-lobby/"))
+					presenceData.details = "In an online lobby";
+				else {
+					switch (document.location.pathname) {
```
Comment (ImRodry) [inline websites/G/Geotastic/presence.ts:74]: a switch inside a switch for the same thing? Come on

File: websites/G/Geotastic/presence.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+							break;
+						}
+						default:
+							if (document.location.pathname.includes("/events/")) {
+								presenceData.details = "Viewing an event";
+								presenceData.state = document.querySelector("h1").textContent;
+							} else if (document.location.pathname.includes("/help-out/"))
+								presenceData.details = "Viewing ways to support Geotastic";
+							else {
+								switch (document.location.pathname) {
```
Comment (ImRodry) [inline websites/G/Geotastic/presence.ts:110]: again... let's not do this please

File: websites/G/Geotastic/presence.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+									}
+									case "/account/quota-consumption": {
+										presenceData.details = "Viewing their quota consumption";
+										break;
+									}
+									case "/account/account-settings":
+										{
+											presenceData.details = "Viewing their account settings";
+										}
+										break;
```
Comment (ImRodry) [inline websites/G/Geotastic/presence.ts:139]: why did this one decide to be the quirky young sibling

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/G/Geotastic/presence.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+			break;
+		}
+		case "/account/quota-consumption": {
+			presenceData.details = "Viewing their quota consumption";
+			break;
+		}
+		case "/account/account-settings": {
+			presenceData.details = "Viewing their account settings";
+			break;
+		}
```
Comment (EncryptedDev) [inline websites/G/Geotastic/presence.ts:93]: Please use a Record for this instead as shown below. Much shorter and more maintainable 
https://github.com/PreMiD/Presences/blob/113adabd2296ec1be22122ccc3ec2deff1c399ed/websites/T/TruckersMP/presence.ts#L15-L41

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by itsmelouis — APPROVED** [review]

**Review by ImRodry — APPROVED** [review]

**Pull request merged**